### DISCUSSION TITLE
How to measure total cell's capacity fade?

### DISCUSSION THREAD

**Discussion by cyasing** [pybamm-team/PyBaMM#2838] [Q&A]
I tried these three ways of doing it with default model and parameters and SPM:
1.  `capacity_fade_val[i] = (solution.summary_variables["Measured capacity [A.h]"][0] 
                    - solution.summary_variables["Measured capacity [A.h]"][-1]) / solution.summary_variables["Measured capacity [A.h]"][0]`
2. `capacity_fade_val[i] = (param["Nominal cell capacity [A.h]"] -(p+n)) / param["Nominal cell capacity [A.h]"] `
 where p and n are: `p = solution.summary_variables['Positive electrode capacity [A.h]'][-1]`
`n = solution.summary_variables['Negative electrode capacity [A.h]'][-1]`
3. `capacity_fade_val[i] = (param["Nominal cell capacity [A.h]"] - solution.summary_variables['Total capacity lost to side reactions [A.h]'][-1]) / param["Nominal cell capacity [A.h]"] 
`
but 1 and 2 result in negative values with a discharge (till 3.3 V) and charge (till 4.1 V) experiment, while 3 results in all 1's, i.e. without any losses. What should I do?

**Comment by RuiheLi** [comment]
Hi, some ageing mechanisms must be enabled to make the cell degrade, for example: `spm = pybamm.lithium_ion.SPM({"SEI": "ec reaction limited"})`, you can check a notebook [here](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/simulating-long-experiments.ipynb). Also you would need to extract a discharge step rather than the whole solution. I think the first one is closer to the right one, though I personally prefer the variable "Discharge capacity [A.h]"

**Reply by cyasing** [reply]
Of course, I know about the options list. It's quite strange though that the capacity does not act normally like other variables.
